### PR TITLE
rework tests and add benchmarks

### PR DESCRIPTION
**Pull request by artofnothingness**

### CODE REVIEW

**Comment by artofnothingness** [comment]
@SteveMacenski 

Benchmarks shows not more than 5 ms on my machine for optimizer evalControl (with all critics enabled)

Controller does three things 
1) evaluate next control (optimizer evalControl)
2) transform path (path handler) take less than 1 ms
3) visualize 

Can you analyse what could be wrong? Mb something with benchmarks ?

File: package.xml [hunk 1]
```diff
@@ -25,6 +25,7 @@
   <depend>tf2_eigen</depend>
   <depend>tf2_ros</depend>
   <depend>std_msgs</depend>
+  <depend>libbenchmark-dev</depend>
```
Comment (SteveMacenski) [inline package.xml]: its just `benchmark` as the rosdep key that maps to that binary name

**Comment by SteveMacenski** [comment]
What about from the plugin API for control https://github.com/artofnothingness/mppic/blob/develop/src/controller.cpp#L54 so we have everything included that this controller is doing (path handler, visualization, etc)? Just to get those metrics as well. 

I don't know off hand, I'd first try to get print timestamps when you run it normally (launch whole thing in simulation) on start/end of `computeVelocityCommand` API at runtime and see if that is consistent with what you see with benchmarks. If the actual run-time is that low, then its worth looking at where the overhead is that's throttling this. But if it is indeed that small and there's some overhead from the `controller_server`, then there should be no reason we couldn't iterate many, many times since that `controller_server` overhead is static.

**Comment by artofnothingness** [comment]
> What about from the plugin API for control https://github.com/artofnothingness/mppic/blob/develop/src/controller.cpp#L54 so we have everything included that this controller is doing (path handler, visualization, etc)? Just to get those metrics as well.

Path handler less then 1 ms as i said before. I'ts hard to make controller benchmarks work without canTrasfrom errors. Some of them work's others not. Visualization is not so interesting now

**Comment by artofnothingness** [comment]
added controller benchmark, as i said, it almost the same time. not all cases works in this benchmarks coz of tf can trasform errors, so if you know hot to fix it, open pr plz

**Pull request merged**